game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/AggregateResults.java

```diff
@@ -107,103 +195,362 @@ public double getAverageTuvSwing(
-    }
-    if (total <= 0) {
-      return 0;
-    }
-    return count / total;
+    final int attackerStartingTuv = TuvUtils.getTuv(attackers, attackerCostsForTuv);
+    final int defenderStartingTuv = TuvUtils.getTuv(defenders, defenderCostsForTuv);
+    final Mean mean = new Mean();
+    final double m =
+        mean.evaluate(
+            results.stream()
+                .mapToDouble(
+                    result ->
+                        TuvUtils.getTuv(result.getRemainingAttackingUnits(), attackerCostsForTuv)
+                            - TuvUtils.getTuv(
+                                result.getRemainingDefendingUnits(), defenderCostsForTuv))
+                .toArray());
+    // If there are no battles, stdDev.getResult() returns Double.NaN.
+    return Double.isNaN(m) ? 0 : defenderStartingTuv - attackerStartingTuv + m;
   }
 
-  double getAverageDefendingUnitsLeft() {
-    if (results.isEmpty()) {
-      return 0.0;
-    }
-    return results.stream()
-            .map(BattleResults::getRemainingDefendingUnits)
-            .mapToDouble(Collection::size)
-            .sum()
-        / results.size();
+  /**
+   * Returns the standard deviation of the TUV swing.
+   *
+   * <p>Note: The result of this function only makes sense for battle with the same initial setup,
+   * e.g. results coming from simulations of the same battle. The {@code attackers} and {@code
+   * defenders} are the units attacking and defending in this setup, respectively.
+   *
+   * <p>{@code attacker}, {@code defender} and {@code data} are used to derive the TUV values of the
+   * units.
+   *
+   * @param attacker the attacking player
+   * @param defender the defending player
+   * @param data the game data
+   * @return the standard deviation for the TUV swing of the battles.
+   */
+  public double getAverageTuvSwingStandardDeviation(
+      final GamePlayer attacker, final GamePlayer defender, final GameData data) {
+    // The variance is invariant under a constant shift.  Thus, the variance of
+    // the TUV of left-over units and the variance of the TUV of lost units is
+    // the same as they differ only be the (constant) total TUV the units at
+    // the start of the battle.
+    final IntegerMap<UnitType> attackerCostsForTuv = TuvUtils.getCostsForTuv(attacker, data);
+    final IntegerMap<UnitType> defenderCostsForTuv = TuvUtils.getCostsForTuv(defender, data);
+    // The 'true' argument means calculate the variance using bias correction.
+    final StandardDeviation stdDev = new StandardDeviation(true);
+    final double s =
+        stdDev.evaluate(
+            results.stream()
+                .mapToDouble(
+                    result ->
+                        TuvUtils.getTuv(result.getRemainingAttackingUnits(), attackerCostsForTuv)
+                            - TuvUtils.getTuv(
+                                result.getRemainingDefendingUnits(), defenderCostsForTuv))
+                .toArray());
+    return Double.isNaN(s) ? 0 : s;
```

Comment:
This block is slow to grok. Would you mind looking to see if it can be made 'lighter' to understand?
Suggestions:
- spell out the variable 's'
- put a comment above the stream operation to give a short summary of what we're doing
- consider using intermediate variables so there is less computation per line of code

Reply:
Let's see what we can do.  The sad part is, that the auto formatting breaks the symmetry between attacker TUV and defender TUV, otherwise you would instantly see that the middle lines are the same but with attacker changed into defender.